### PR TITLE
Add Terraform Template language

### LINKED ISSUE #6562: New language: Terraform templates
## Language name

Terraform templates

## URL of example repository

Not sure, but there are examples in the docs, linked below.

## URL of syntax highlighting grammar

https://developer.hashicorp.com/terraform/language/v1.5.x/expressions/strings#string-templates

## Most popular extensions

`.tftpl` is [recommended](https://developer.hashicorp.com/terraform/language/v1.5.x/functions/templatefile)

## Detected language

None

### PR DESCRIPTION
**Pull request by johnmays**
## Description

Closes #6562

I added Terraform Template as a new markup language.  Terraform template files are used to automatically add syntax when given a set of arguments in Terraform.  See the [Terraform documentation page](https://developer.hashicorp.com/terraform/language/functions/templatefile) for more information.

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  [~1.8k search results](https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.tftpl) for the single extension: `.tftpl`
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [`donors.tftpl`](https://github.com/GoogleCloudPlatform/emblem/blob/38ae907f2a82043a8ff025d0da742037b4d50ae1/terraform/modules/emblem-app/files/templates/donors.tftpl#L4)
      - ~~[`os_alpine_3_17_userdata.tftpl`](https://github.com/k0sproject/k0s/blob/a4253d153a7fba5e334e280ab646bd8689ce2f05/hack/ostests/modules/os/os_alpine_3_17_userdata.tftpl#L4)~~ *(removed in 7ce7db667ca41f60d02e6fa97c6b4ac46d7495bc)*
      - [`policy.tftpl`](https://github.com/omerbsezer/Fast-Terraform/blob/f465f91e44f82a5792c7fe2ee56bff899f0abadf/labs/template/policy.tftpl#L4)
    - Sample license(s):
      - [Apache 2.0](https://github.com/GoogleCloudPlatform/emblem/blob/38ae907f2a82043a8ff025d0da742037b4d50ae1/LICENSE)
      - ~~[Apache 2.0](https://github.com/k0sproject/k0s/blob/a4253d153a7fba5e334e280ab646bd8689ce2f05/LICENSE)~~
      - [MIT](https://github.com/omerbsezer/Fast-Terraform/blob/f465f91e44f82a5792c7fe2ee56bff899f0abadf/LICENSE)
  - [x] I have included [a syntax highlighting grammar](https://github.com/hashicorp/syntax/blob/main/syntaxes/terraform.tmGrammar.json) ([uses MPL 2.0](https://github.com/hashicorp/syntax/blob/main/LICENSE))
      
  - [x] I have added a color
    - Hex value: `#7B42BB`
    - Rationale: It is one of the purples Hashicorp, the developers of Terraform, officially use for it. The purple in use for the Terraform (AKA "HCL") language is the other purple.  It is similar, but noticeably brighter.  They have these two official purples listed on in their [brand page for Terraform](https://www.hashicorp.com/brand). 
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.
    - No other languages use `.tftpl`, so there is no need.

## Misc. Note:
I did list the `Terraform Template` Language in the `HCL` group. (`HCL` is the primary alias for `Terraform`).  This will count it as `HCL` instead of its own language.  I think this is appropriate, but I'm  not 100% sure.  I'm also not sure if that dismisses the need for a color choice.

### CODE REVIEW

File: samples/Terraform Template/os_alpine_3_17_userdata.tftpl [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
```
Comment (lildude) [inline samples/Terraform Template/os_alpine_3_17_userdata.tftpl:1]: This is going to cause troubles as this is a shell interpreter not a Terraform interpreter and we can't have two languages sharing an interpreter. It's probably best to remove this sample entirely.

**Review by lildude — CHANGES_REQUESTED** [review]
Please address the test failed. My comment inline will deal with the interpreter/shell failures.

File: samples/Terraform Template/os_alpine_3_17_userdata.tftpl (hunk 1 above)
Comment (johnmays) [reply samples/Terraform Template/os_alpine_3_17_userdata.tftpl:1]: @lildude Oh yes you're right.  I'll look at & fix tomorrow.  Thanks.

**Comment by lildude** [comment]
> > Please address the test failed. My comment inline will deal with the interpreter/shell failures.
> 
> I removed the sample, ran the tests locally (they passed), and pushed. Sorry, I am kind of new to this, but should I resolve the conflicts in `.gitmodules` or will that be addressed when you merge?

It's best if you do it

File: .gitmodules [hunk 2]
```diff
@@ -1157,6 +1157,12 @@
 [submodule "vendor/grammars/sweave.tmbundle"]
 	path = vendor/grammars/sweave.tmbundle
 	url = https://github.com/textmate/sweave.tmbundle.git
+[submodule "vendor/grammars/swift.tmbundle"]
+	path = vendor/grammars/swift.tmbundle
+	url = https://github.com/textmate/swift.tmbundle
```
[the remark this answers is not in the document]
Comment (johnmays) [reply .gitmodules:1162]: Okay.  The tests will not run unless I have a grammar, so should I rebase to have this updated one?

**Comment by lildude** [comment]
Sorry about the noise. Something really went awry with the submodules and cached licenses which took a while to get right, but we should be good to go now. CI should pass now 🤞

**Comment by johnmays** [comment]
> Sorry about the noise. Something really went awry with the submodules and cached licenses which took a while to get right, but we should be good to go now. CI should pass now 🤞

All good.  Very nice to see.

**Pull request merged**